### PR TITLE
Creating initial tsb role to consume and apply templates provided for…

### PR DESCRIPTION
**Pull request by ewolinetz**
… tsb

cc: @deads2k @sdodson 

Addresses:
https://bugzilla.redhat.com/show_bug.cgi?id=1486623
https://bugzilla.redhat.com/show_bug.cgi?id=1470623
https://bugzilla.redhat.com/show_bug.cgi?id=1491626

### CODE REVIEW

**Comment by ewolinetz** [comment]
Changes dependent on https://github.com/openshift/origin/pull/15993

**Comment by ewolinetz** [comment]
@deads2k is this the correct way to handle the TSB config? I'm seeing this when I try to `kubectl apply` it

```json
fatal: [m01.example.com]: FAILED! => {
    "changed": true, 
    "cmd": [
        "kubectl", 
        "apply", 
        "-f", 
        "/tmp/tsb-ansible-lS4vzn/apiserver-config.yaml"
    ], 
    "delta": "0:00:00.219908", 
    "end": "2017-09-08 16:51:36.256954", 
    "failed": true, 
    "invocation": {
        "module_args": {
            "_raw_params": "kubectl apply -f /tmp/tsb-ansible-lS4vzn/apiserver-config.yaml", 
            "_uses_shell": false, 
            "chdir": null, 
            "creates": null, 
            "executable": null, 
            "removes": null, 
            "warn": true
        }
    }, 
    "rc": 1, 
    "start": "2017-09-08 16:51:36.037046", 
    "stderr": "error: unable to recognize \"/tmp/tsb-ansible-lS4vzn/apiserver-config.yaml\": no matches for config.templateservicebroker.openshift.io/, Kind=TemplateServiceBrokerConfig", 
    "stderr_lines": [
        "error: unable to recognize \"/tmp/tsb-ansible-lS4vzn/apiserver-config.yaml\": no matches for config.templateservicebroker.openshift.io/, Kind=TemplateServiceBrokerConfig"
    ], 
    "stdout": "", 
    "stdout_lines": []
}
```

**Comment by deads2k** [comment]
I'm afraid I don't speak ansible very well. You're doing `oc process -f install/templateservicebroker/apiserver-template.yaml | oc apply -f -` or equivalent?

**Comment by ewolinetz** [comment]
@deads2k apologies... I'm doing `kubectl apply -f /tmp/tsb-ansible-lS4vzn/apiserver-config.yaml`  should I do `oc process -f .... | oc apply -f -` ?  My server returned saying that it doesn't recognize the object kind...

```no matches for config.templateservicebroker.openshift.io/, Kind=TemplateServiceBrokerConfig```

**Comment by deads2k** [comment]
> I'm doing kubectl apply -f /tmp/tsb-ansible-lS4vzn/apiserver-config.yaml

I don't recognize that file.  Is it a template file or a processed template (list)?  If it is https://github.com/openshift/origin/blob/master/install/templateservicebroker/apiserver-config.yaml , then that's the config file for the server which you will pass to the template as a parameter.  You don't apply that one.

**Comment by ewolinetz** [comment]
@deads2k ack, yes... I'm forgetting that there's that layer of abstraction. It is that config file... so when I create the server template I pass this as a parameter, got it! Much appreciated

**Comment by ewolinetz** [comment]
@sdodson @deads2k any qualms with separating this work from the work required for https://github.com/openshift/openshift-ansible/issues/5056 ?

**Comment by sdodson** [comment]
> @sdodson @deads2k any qualms with separating this work from the work required for #5056 ?

No that's fine with me. #5056 would be addressed first?

**Comment by ewolinetz** [comment]
They'd both need to be delivered for 3.7 but it might make sense to get #5056 merged in first

File: roles/template_service_broker/tasks/remove.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
```
Comment (kwoodson) [inline roles/template_service_broker/tasks/remove.yml]: What do we do when any of these commands fail?  How do we ensure that these objects are removed?  Just something to think about.

**Comment by ewolinetz** [comment]
aos-ci-test

**Comment by ewolinetz** [comment]
@deads2k with the latest tested locally I see the TSB pods start up and run after an installation

**Comment by deads2k** [comment]
@sdodson @kwoodson I haven't seen much review action on this.  I don't know enough to meaningfully review it, but if it can get reviewed in time to prove the pattern you may be able to use it for something like service catalog this release too.

File: roles/openshift_sanitize_inventory/tasks/__deprecations_metrics.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    openshift_metrics_image_version: "{{ openshift_hosted_metrics_deployer_version }}"
+  when: openshift_hosted_metrics_deployer_version is defined
+
+- set_fact:
+    openshift_metrics_install_metrics: "{{ openshift_hosted_metrics_deploy }}"
+  when: openshift_hosted_metrics_deploy is defined
+
+- set_fact:
+    openshift_metrics_storage_kind: "{{ openshift_hosted_metrics_storage_kind }}"
+  when: openshift_hosted_metrics_storage_kind is defined
```
Comment (sdodson) [inline roles/openshift_sanitize_inventory/tasks/__deprecations_metrics.yml]: This file seems unrelated to the topic of the PR?

File: roles/template_service_broker/tasks/install.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  changed_when: False
+  become: no
+
+- name: Pull down files
+  command: >
+    wget {{ __tsb_files_location }}{{ item }} -P {{ mktemp.stdout }}
+  with_items:
+    - "{{ __tsb_template_file }}"
+    - "{{ __tsb_config_file }}"
+    - "{{ __tsb_rbac_file }}"
```
Comment (sdodson) [inline roles/template_service_broker/tasks/install.yml]: Has this stabilized enough that we should go ahead and add this content to openshift-ansible in the location we agreed upon rather than fetching it from origin github?

File: roles/template_service_broker/tasks/remove.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  command: >
+    wget {{ __tsb_files_location }}{{ __tsb_template_file }} -P {{ mktemp.stdout }}
+
+- name: Pull down config
+  command: >
+    wget {{ __tsb_files_location }}{{ __tsb_config_file }} -P {{ mktemp.stdout }}
+
+- name: Delete template file
+  command: >
+    kubectl delete -f {{ mktemp.stdout }}/{{ __tsb_template_file }}
```
Comment (sdodson) [inline roles/template_service_broker/tasks/remove.yml]: kubectl versus oc in the next task, is that intentional?

Comment (sdodson) [reply roles/template_service_broker/tasks/remove.yml]: @deads2k delete on a template will delete all the resources defined in the template?

**Comment by sdodson** [comment]
Assuming the metrics related stuff is unrelated to this PR lets remove that then I'm LGTM.

File: roles/openshift_sanitize_inventory/tasks/__deprecations_metrics.yml (hunk 2 above)
Comment (ewolinetz) [reply roles/openshift_sanitize_inventory/tasks/__deprecations_metrics.yml]: Ugh, i committed while on the wrong branch locally... but i thought i reverted it. Will remove

File: roles/template_service_broker/tasks/install.yml (hunk 3 above)
Comment (ewolinetz) [reply roles/template_service_broker/tasks/install.yml]: I thought the alternative to pulling the content from origin was going to be a locally sync'd place in ansible. Is that configured yet? We should be able to just change the source location in `vars/`

Comment (sdodson) [reply roles/template_service_broker/tasks/install.yml]: Yeah, we'll setup CI tooling for that but lets go ahead and import a copy of current content as part of this PR?

File: roles/template_service_broker/tasks/remove.yml (hunk 4 above)
Comment (deads2k) [reply roles/template_service_broker/tasks/remove.yml]: > @deads2k delete on a template will delete all the resources defined in the template?

You need to process the template, then pipe that to delete.  You probably want to delete the entire namespace (after doing this) just in case.

Comment (sdodson) [reply roles/template_service_broker/tasks/remove.yml]: I was going to suggest just nuking the namespace.

Comment (deads2k) [reply roles/template_service_broker/tasks/remove.yml]: > I was going to suggest just nuking the namespace.

both are useful.  Some resources which are created are cluster scoped.  ClusterRoles for instance.

**Comment by ewolinetz** [comment]
@sdodson @deads2k how does everything look?

**Comment by ewolinetz** [comment]
Addresses:
https://bugzilla.redhat.com/show_bug.cgi?id=1486623
https://bugzilla.redhat.com/show_bug.cgi?id=1470623
https://bugzilla.redhat.com/show_bug.cgi?id=1491626
